pkg/httphelper/httphelper.go

```diff
@@ -6,14 +6,30 @@ import (
 	"net/http"
 )
 
-func Error(w http.ResponseWriter, err error) {
+type JSONError struct {
+	Code    int         `json:"code"`
```

Comment:
Agree with having a string, enumerated somewhere.

Perhaps we try to keep them generic APIerrors as long as we can and revist component namespaces later if we find overwhelming need?  So far the most major itches I've all seem to be general ones like "API not found" vs "ID not found".  "object exists" is also a good example (I think @archSeer metioned in another thread that that maps nicely on to http 409, but it a json chuck clarifying that the "conflict" is because "exists" makes total sense).